pandas/core/groupby/groupby.py

```diff
@@ -2675,6 +2674,51 @@ def rank(
         Returns
         -------
         DataFrame with ranking of values within each group
+
+        See Also
+        --------
+        Series.groupby : Apply a function groupby to a Series.
+        DataFrame.groupby : Apply a function groupby
+            to each row or column of a DataFrame.
+        Series.rank : Apply a function rank to a Series.
+        DataFrame.rank : Apply a function rank
+            to each row or column of a DataFrame.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({'group': ['a', 'a', 'a', 'b',
+        ...                              'a', 'b', 'b', 'b', 'b', 'a'],
+        ...                    'value': [.2, .4, .2, 0.01,
+        ...                              .3, .11, .21, .4, .01, 0.2]})
+        >>> df
+          group  value
+        0     a   0.20
+        1     a   0.40
+        2     a   0.20
+        3     b   0.01
+        4     a   0.30
+        5     b   0.11
+        6     b   0.21
+        7     b   0.40
+        8     b   0.01
+        9     a   0.20
```

Comment:
I think the example would be easier to see how different groups are treated if groups are contiguous, eg a, a, a, a...b, b, b, b

Reply:
Also think it would be clearer to have fewer distinct values (and maybe use ints instead of floats, with values that are easy to tell at a glance what is smallest, largest, etc